api/R/get_model_id.R

```diff
@@ -0,0 +1,37 @@
+#' Retrieve database ID of a particular version of a model
+#'
+#' @param con Database connection object (Pqconnection)
+#' @param name Model name (character)
+#' @param revision Model version/revision (character)
+#' @param multi_action Action to take if multiple models found
+#'   (character). Must be one of "first", "last" (default), "all", or "error".
+#' @return Model ID, as `integer64`
+#' @author Alexey Shiklomanov
+#' @export
+get_model_id <- function(con, name, revision, multi_action = "last") {
+  qry <- DBI::dbSendQuery(con, paste(
+    "SELECT id FROM models WHERE model_name = $1 and revision = $2 ORDER BY id DESC"
```

Comment:
It's the very next line. Look at [`?DBI::dbBind`](https://www.rdocumentation.org/packages/DBI/versions/0.5-1/topics/dbBind).

Reply:
I really love that, we should work to replace some of our function calls with DBI::dbBind and parameterized functions. This would get around some SQL injections.

Reply:
@robkooper Agree. Re-writing all the PEcAn.DB functions to use this has been on the back of my mind for a while.

An important caveat is that these kinds of prepared statements are _not_ supported by the `RPostgreSQL` package that we currently use. We would have to switch to the newer, fully DBI-compliant `RPostgres` package (which I use throughout this package). In theory, this should just be a quick find-replace, but it may break backwards-compatibility for a lot of old XML files that still have `driver = PostgreSQL`. I vote we add this to the list of PEcAn 2.0 features, along with deprecating the `driver` tag altogether (since we only support PostgreSQL).

Reply:
Ahh, I get it. I didn't catch the distinction between "SendQuery" and "Fetch" -- it looked like you were submitting the query to the DB before doing the substitution

Reply:
I agree that the function name is a bit confusing. `sendQuery` definitely sounds like it should be actually executing the query, and in some sense, it does, but AFAICT, "sending" is not fully the same as "processing" from PostgreSQL's point of view.